examples/applications/author-disambiguation/sampling.py

```diff
@@ -0,0 +1,169 @@
+                        if y_train[i] == y_train[j] and name_i == name_j:
+                            b_sasn.append((sig_i, sig_j, 0))
+                        if y_train[i] == y_train[j] and name_i != name_j:
+                            b_sadn.append((sig_i, sig_j, 0))
+                        if y_train[i] != y_train[j] and name_i != name_j:
+                            b_dadn.append((sig_i, sig_j, 1))
+
+        if len(b_sadn) > k:
+            s_sadn = \
+                [b_sadn[i] for i in random.sample(range(len(b_sadn)), k)]
+        else:
+            s_sadn = b_sadn
+        if len(b_sasn) > k:
+            s_sasn = \
+                [b_sasn[i] for i in random.sample(range(len(b_sasn)), k)]
+        else:
+            s_sasn = b_sasn
+        if len(b_dadn) > k:
+            s_dadn = \
+                [b_dadn[i] for i in random.sample(range(len(b_dadn)), k)]
+        else:
+            s_dadn = b_dadn
+
+        sasn = sasn + s_sasn
+        sadn = sadn + s_sadn
+        dadn = dadn + s_dadn
+
+    if balanced:
+        dasn_s = [dasn[i] for i in random.sample(range(len(dasn)), s)]
+        sadn_s = [sadn[i] for i in random.sample(range(len(sadn)), s)]
+        sasn_s = [sasn[i] for i in random.sample(range(len(sasn)), s)]
+        dadn_s = [dadn[i] for i in random.sample(range(len(dadn)), s)]
+
+        pairs = dasn_s + sadn_s + sasn_s + dadn_s
+    else:
+        pairs_l = dasn + sadn + sasn + dadn
+        rand_sampl = random.sample(range(len(pairs_l)), sample_size)
+        pairs = [pairs_l[i] for i in rand_sampl]
+
+    return pairs
+
+if __name__ == "__main__":
+    # Parse command line arugments
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--output_pairs", default="pairs.pickle", type=str)
+    parser.add_argument("--input_signatures", default="signatures.json", type=str)
+    parser.add_argument("--input_records", default="records.cPickle", type=str)
+    parser.add_argument("--input_clusters", default="clusters.json", type=str)
+    parser.add_argument("--input_blocking", default=block_last_name_first_initial, type=str)
+    parser.add_argument("--input_test_size", default=0.9, type=float)
+    parser.add_argument("--input_sample_size", default=10000, type=int)
+    parser.add_argument("--input_random_state", default=42, type=int)
+    parser.add_argument("--input_balanced", default=True, type=bool)
+
+    args = parser.parse_args()
+
+    pairs = sampling_pairs(blocking_func=args.input_blocking,
+                           sample_size=args.input_sample_size,
+                           signatures_filename=args.input_signatures,
+                           records_filename=args.input_records,
+                           clusters_filename=args.input_clusters,
+                           test_size=args.input_test_size,
+                           random_state=args.input_random_state,
+                           balanced=args.input_balanced)
+
+    pickle.dump(pairs, open(args.output_pairs, "w"))
```

Comment:
Please dump to json. The disadvantage of a pickle is that it is harder to make it compatible between Python2 and Python3.